eogrow/utils/testing.py

```diff
@@ -163,37 +165,33 @@ def _calculate_numpy_stats(raster: np.ndarray, config: StatCalcConfig) -> JsonDi
 
 def _calculate_tiff_stats(tiff_filename: str, config: StatCalcConfig) -> JsonDict:
     """Calculates statistics over a .tiff image"""
-    with open(tiff_filename, "rb") as file_handle:
-        with rasterio.open(file_handle) as tiff:
-            image = tiff.read()
-            mask = tiff.dataset_mask()
-
-    return {
-        "image": _calculate_numpy_stats(image, config),
-        "mask": _calculate_numpy_stats(mask, config),
-    }
-
-
-def _calculate_numpy_file_stats(numpy_filename: str, config: StatCalcConfig) -> JsonDict:
-    """Calculates statistics over a .npy file containing a numpy array"""
-    with open(numpy_filename, "rb") as file_handle:
-        raster = np.load(file_handle, allow_pickle=True)
-
-    return _calculate_numpy_stats(raster, config)
+    with rasterio.open(tiff_filename) as tiff:
+        return {
+            "image": _calculate_numpy_stats(tiff.read(), config),
+            "mask": _calculate_numpy_stats(tiff.dataset_mask(), config),
+        }
 
 
-def _calculate_vector_stats(dataframe: pd.DataFrame, config: StatCalcConfig) -> JsonDict:
+def _calculate_vector_stats(dataframe: gpd.GeoDataFrame, config: StatCalcConfig) -> JsonDict:
     """Calculates statistics over a vector GeoDataFrame"""  # TODO: add more statistical properties
 
     def _rounder(x: float, y: float) -> tuple[float, float]:
         return round(x, config.decimals), round(y, config.decimals)
 
-    dataframe["geometry"] = dataframe["geometry"].apply(lambda geometry: shapely.ops.transform(_rounder, geometry))
+    dataframe.geometry = dataframe.geometry.apply(lambda geometry: shapely.ops.transform(_rounder, geometry))
 
-    stats = {"columns": list(dataframe), "row_count": len(dataframe.index), "crs": str(dataframe.crs)}
+    stats = {
+        "columns": list(dataframe),
+        "row_count": len(dataframe.index),
+        "crs": str(dataframe.crs),
+        "mean_area": _prepare_value(dataframe.area.mean(), config),
+        "total_bounds": list(dataframe.total_bounds),
+    }
 
     if len(dataframe.index):
-        stats["first_row"] = list(map(str, dataframe.iloc[0]))
+        rng = np.random.default_rng(0)
+        random_rows = np.unique(rng.integers(0, len(dataframe), config.num_random_values))
+        stats["random_rows"] = {str(row): list(map(str, dataframe.iloc[row])) for row in random_rows}
```

Comment:
Additionally, I agree regarding your comment about saving  the full geometries in the random rows.

You could add the centroid column after L181 (or even later, after L191):

```python
dataframe["centroid"] = dataframe.geometry.centroid
```

and then after the check in L191 just drop the geometry column, so that it's not included in the random rows stat

Reply:
I added the centroid and area columns. The dataframe sampling seems neat, though i had to be a bit careful with it. It gives you elements that are not inherently jsonifiable, so i do a quick to-json-and-back transition before comparing stats.

i also decided that for each polygon the first 10 coordinates are saved. kinda want at least some (centroid and area cannot detect rotations etc.)

Reply:
Is the issue just the jsonifability, or did you in addition prepare the values to be of a smaller size? If the latter is not necessary, there is also `to_json(orient='index')`, which could be used to drop a direct json, or you coud wrap it in `json.loads(...)` to get a jsonifiable dict. But that's feels weird to do.

Reply:
`to_json` saves the file, i just need it 'as if it were loaded from a json' so i do `json.loads(json.dumps(x))`

Reply:
if you don't provide the filename it also just provides the string

Reply:
I see.
I would then need to also `json.load` it, yeah. It's similarly 'nasty' than what is currently done. I can switch if you'd prefer this.

Reply:
no preference, just wanted to mention it in case you preferred it.